### PR TITLE
Child only resource discovery

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Configure child only resource discovery to avoid discovering certain resources across class loaders.

## Type
 - [x] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
For certain resources such as `org/slf4j/impl/StaticLoggerBinder.class`, it does not make sense to discovery a parent one because it will almost certain cause classloading problem when mixing parent class and child class.

## Tracking Issue
Closes https://github.com/flyteorg/flyte/issues/3051

## Follow-up issue
_NA_

### CODE REVIEW

File: jflyte/src/main/java/org/flyte/jflyte/ChildFirstClassLoader.java [hunk 1]
```diff
@@ -42,6 +45,9 @@ class ChildFirstClassLoader extends URLClassLoader {
   private static final String[] PARENT_FIRST_PACKAGE_PREFIXES =
       new String[] {"org.flyte.api.v1.", "org.flyte.jflyte.api."};
 
+  private static final Set<String> CHILD_ONLY_RESOURCES =
```
Comment (honnix) [inline jflyte/src/main/java/org/flyte/jflyte/ChildFirstClassLoader.java:48]: This project is still targeting Java 8, so no factory method from `Set` to use.

**Review by rafaelraposospot — APPROVED** [review]

**Comment by honnix** [comment]
I don't have permission to merge. Maybe I should be added to some team?
